widelands_pre-install.bash

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+cd "$( dirname "$( readlink -f "${0}" )" )";
+source './widelands_setup.bash';
+
+echo 'create icons';
+icon="${app_name}.svg";
+for s in {16,22,24,32,48,64,72,96,128,192,256,512}; do
+  size="${s}x${s}";
+  echo "- ${size}";
+  mkdir -p "icons/${size}/";
+  rsvg-convert "${icon}" -w "${s}" -h "${s}" -a -f png -o "icons/${size}/${app_name}.png";
+done;
+echo;
+
+echo 'create desktop';
+cp "debian/${app_id}.desktop" "${app_name}.desktop";
+echo;
+
+echo 'create appdata';
+cp "debian/${app_name}.appdata.xml" "${app_name}.appdata.xml";
+echo;
+
+echo 'edit desktop';
+desktop-file-edit --set-key='Icon' --set-value="${app_name}" "${app_name}.desktop";
+echo;
+
+echo 'edit appdata';
```

Comment:
Just patch appdata file, it's more readable.

Reply:
> Just patch appdata file, it's more readable.

No, it's not. Moreover, I'm not going to create a patch for every single release. And this is necessary, because appstream-util can't correctly compare version numbers (e.g. "Build 9", "Build 9half", "Build 10"). You can say that it is the vendor's fault (especially "Build 9half"), but even Blender had a similar problem in the past.

Reply:
Upstreaming changes to appdata is de facto requirement, unless development has stalled. Bundling xmlstarlet (which isn't needed by the game anyway) does not look like it.

Reply:
You already know that at the end all unnecessary elements are removed:
https://github.com/scx/flathub/blob/4b76d59262f97c3072304f13dbe4a3e1fc2be742/org.widelands.widelands.yaml#L30-L40

Reply:
You should use `cleanup: ["*"]` instead assuming it won't put files outside the prefix; conveniently you ignored the rest of my comment.

Reply:
> You should use `cleanup: ["*"]` instead

As you said before: "that's fine too".

> conveniently you ignored the rest of my comment.

This game has not been officially abandoned, but the latest version was released two years ago. And I don't think the vendor would be willing to change the version naming scheme just because of Flatpak/Flathub.